gui/job-details.lua

```diff
@@ -0,0 +1,438 @@
+-- Show and modify properties of jobs in a workshop or of manager orders.
+
+--[[
+    Credit goes to the author of `gui/workshop-job`.
+    This is the gui/workshop-job.lua with gui/workorder-details.lua added on top
+    and updated to the DF50.
+]]
+
+--@ module = true
+
+local overlay = require('plugins.overlay')
+
+local utils = require 'utils'
+local gui = require 'gui'
+local guimat = require 'gui.materials'
+local widgets = require 'gui.widgets'
+local dlg = require 'gui.dialogs'
+
+JobDetails = defclass(JobDetails, gui.ZScreenModal)
+
+JobDetails.focus_path = 'job-details'
+
+JobDetails.ATTRS {
+    job = DEFAULT_NIL,
+    context = DEFAULT_NIL,
+    frame_inset = 1,
+    frame_background = COLOR_BLACK,
+}
+
+function JobDetails:isManagerOrder()
+    return self.context == df.job_details_context_type.MANAGER_WORK_ORDER
+end
+
+function JobDetails:init(args)
+    local status
+    if not self:isManagerOrder() then
+        status = { text = 'No worker', pen = COLOR_DARKGREY }
+        local worker = dfhack.job.getWorker(self.job)
+        if self.job.flags.suspend then
+            status = { text = 'Suspended', pen = COLOR_RED }
+        elseif worker then
+            status = { text = dfhack.TranslateName(dfhack.units.getVisibleName(worker)), pen = COLOR_GREEN }
+        end
+    end
+
+    local window = widgets.Window{
+        frame_title='Details',
+        resizable = true,
+        resize_min={w=50, h=20},
+        frame = { l = 10, w = 50 },
+    }
```

Comment:
while this is perfectly valid (and probably the way I did it myself in some earlier scripts), I would rather encourage you to keep the ZScreen and the Window as separate classes. This allows the ZScreen subclass to focus on tool lifecycle and the Window subclass to focus on content and interactive behaviors. In particular, if there is no Window subclass, then future editors might be tempted to add an `onInput` function to the ZScreen subclass, which could break the ZScreen stack.

Reply:
ping on this

Reply:
I'm on it, busy with work and stuff, but I'll get it done